File: archiV1java/inventory/src/main/java/om/inventory/controller/InventoryController.java
package om.inventory.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import om.inventory.svc.InventoryService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController @Slf4j
@RequestMapping("/api/v1/inventory")
public class InventoryController {
    private final InventoryService inventoryService;

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public boolean isInStock(@RequestParam String skuCode, @RequestParam Integer quantityForQuery) {
        log.info("Retrieving inventory stock details for the product SKU: "+ skuCode);
        return inventoryService.isItemInStock(skuCode, quantityForQuery);
    }
}

File: V2kotlin/inventory/src/main/kotlin/om/inventory/controller/InventoryController.kt
package om.inventory.controller

import om.inventory.dto.InventoryItemReq
import om.inventory.service.InventoryService
import org.springframework.http.CacheControl
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.util.concurrent.TimeUnit

@RestController
@RequestMapping("/api/v2/inventory/")
class InventoryController(val inventoryService: InventoryService) {

    @GetMapping
    fun isInStock(@RequestBody inventoryItemReq: InventoryItemReq): ResponseEntity<Boolean> {
        return ResponseEntity.ok()
            .cacheControl(CacheControl.maxAge(30_000, TimeUnit.SECONDS))
            .body(inventoryService.isItemInStock(inventoryItemReq.skuCode, inventoryItemReq.quantityToCheck))
    }

}